src/actions/editorActions.js

```diff
@@ -41,41 +42,41 @@ export function recover() {
 */
 export function fetchScene(id, uid = "anon") {
     return (dispatch) => {  // Return a func that dispatches events after async
-        scenes.doc(id).get().then((scene) => {
-            let data = scene.data();
-            if (data && data.pw) {
-                let pw = prompt("Please enter the PW");
-                if (pw !== data.pw) { return; }
+        fetch(`${sceneRef}/id/${id}`, {redirect: "follow"}).then((response) =>{
+            if(response.redirected && id !== "error-404"){
+                let url = response.url.split("/");
+                window.location.href = window.origin + `/scene/${url[url.length - 1]}`;
+                return;
             }
-            if (data && data.code) { // If it worked
-                // render the editor
-                dispatch(render(data.code, uid || "anon"));
-                dispatch(updateSavedText(data.code));
-                // Use default for eventual consistency in db
-                let settings = DEF_SETTINGS;
 
-                // if the incoming scene has a settings, merge default with incoming
-                if (data.settings) {
-                    settings = { ...settings, ...data.settings };
+            if(response.status !== 200){
+                if(response.status === 404){
+                    window.location.href = window.origin + "/error-404";
```

Comment:
This is probably pulling from old code, but it's better to use window.location.assign(). It's not a big difference, but it's more clear that you're changing the URL and going somewhere.